qiskit/backends/_basebackend.py

```diff
@@ -0,0 +1,40 @@
+"""This module implements the abstract base class for backend modules.
+
+To create add-on backend modules subclass the Backend class in this module. 
+Doing so requires that the required backend interface is implemented.
+"""
+from abc import ABC, abstractmethod
+
+class BaseBackend(ABC):
+    
+    @abstractmethod
+    def __init__(self, qobj):
+        """initialize
+
+        This method should initialize the module and raise a FileNotFoundError 
+        exception if a component of the moduel is not available. 
+
+        Args:
+            qobj (dict): qobj dictionary
+        
+        Raises:
+            FileNotFoundError if backend executable is not available.
+        """
+        self._qobj = qobj
+        self._configuration = None # IMPLEMENT for your backend
+
+    @abstractmethod
+    def run(self):
```

Comment:
Since we have changed to using qobj for communicating tasks the simulator, and perhaps the online backends, don't need to run circuits individually. For instance a c simulator can take in the whole qobj. The run_circuit() method is sort of a non-essential fix to get the python simulators to work with qobj without making too many changes. It seems like a good method to have for python simulators but some simulators might want to do optimizations on batches of circuits which don't need running individual ones. At least this way we make it optional.

Reply:
It sounds sensible, thanks for the clarification - I was mostly not sure how "universal" the `run_circuit()` was meant to be, and keeping it just on the simulator that needs it sounds fine to me!